test/functional/system/transactions/expire_transactions.js

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const liskElements = require('lisk-elements').default;
+const randomUtil = require('../../../common/utils/random');
+const accountsFixtures = require('../../../fixtures/accounts');
+const Scenarios = require('../../common/scenarios');
+const queriesHelper = require('../../common/sql/queriesHelper.js');
+const localCommon = require('../common');
+const Bignum = require('../../../../helpers/bignum.js');
+
+const multiSig = new Scenarios.Multisig();
+
+describe('expire transaction', () => {
+	let library;
+	let queries;
+	let memAccountsSnapshotBeforeUndo;
+
+	const { expiryInterval } = __testContext.config.transactions;
+	const unconfirmedTransactionTimeOut =
+		global.constants.unconfirmedTransactionTimeOut;
+
+	const getSenderAddress = transaction =>
+		transaction.senderId ||
+		library.modules.accounts.generateAddressByPublicKey(
+			transaction.senderPublicKey
+		);
+	const setunconfirmedTransactionTimeOut = timeout => {
+		global.constants.unconfirmedTransactionTimeOut = timeout;
+	};
+
+	// override transaction expiryInterval and unconfirmedTransactionTimeOut
+	// to test undo unConfirmed expired transactions
+	before(done => {
+		__testContext.config.transactions.expiryInterval = 5000;
+		setunconfirmedTransactionTimeOut(0);
+		done();
+	});
+
+	localCommon.beforeBlock('lisk_functional_expire_transactions', lib => {
+		library = lib;
+		queries = new queriesHelper(lib, lib.db);
+	});
+
+	beforeEach('take mem accounts snapshot', () => {
+		return queries.getAccounts().then(accounts => {
+			memAccountsSnapshotBeforeUndo = _.cloneDeep(accounts);
+		});
+	});
+
+	it('should expire transaction from unconfirmed transaction list', done => {
+		const transaction = liskElements.transaction.transfer({
+			recipientId: randomUtil.account().address,
+			amount: randomUtil.number(100000000, 1000000000),
+			passphrase: accountsFixtures.genesis.passphrase,
+		});
+
+		localCommon.addTransactionToUnconfirmedQueue(library, transaction, () => {
+			// Wait for transaction to get expired and
```

Comment:
no expectations in this test?

Reply:
`localCommon.addTransactionToUnconfirmedQueue` doesn't return anything so there are no expectations for that, however, `validateMemAccountSnapshot()` has expectations, let me know if there is anything else that I can try.